### DISCUSSION TITLE
支持一条新的链，比如比特币？

### DISCUSSION THREAD

**Discussion by yutingzhao1991** [ant-design/ant-design-web3#289] [Ideas]
ant design web3 是支持多链的，除了以太坊，感觉还可以官方支持一条其它链，一方面打磨一下多链支持的能力，另外一方面也能吸引更多开发者。

**Comment by yutingzhao1991** [comment]
最近铭文很火，感觉可以支持比特币，支持比特币铭文

**Comment by Likang0122** [comment]
感觉昂贵的手续费会限制生态和开发者0.0

**Reply by yutingzhao1991** [reply]
但是现在铭文真的很火，相比赚的钱，手续费算啥

**Comment by dxiongya** [comment]
支持多链的话，主要麻烦的是如何统一每个不同链的处理逻辑，像starknet，inj等非evm链实际上处理account啥的都不太一样，utils也会有所不同，那么在组件层面和逻辑层面咋规划合适

**Reply by yutingzhao1991** [reply]
可以看看 https://preview-369-ant-design-web3.surge.sh/zh-CN/guide/adapter 这里。

组件目前是和链无关的，加一个其它链的 adapter 就可以了。

acount 目前组件就定义的很简单的格式：

```
{
  name?: string;
  address: string;
}
```

对于组件来说，链直接的差异应该是比较小的，可以适配。

**Reply by dxiongya** [reply]
嗯，昨天凌晨太晚，没细看，看到啦～

**Comment by alienzhangyw** [comment]
有没有计划支持波场和TronLink钱包？

**Reply by jeasonstudio** [reply]
如果是有具体的需求，我们可以提高下优先级～

**Reply by yutingzhao1991** [reply]
@alienzhangyw  有兴趣来 PR 支持么，欢迎 PR~

**Reply by alienzhangyw** [reply]
波场上的USDT使用率更高吧。
我们自己的项目目前是用的connectkit和tronwallet-adapters来连接钱包的，但是手机上的体验都不怎么好，正在考虑重做。

另外我自己写了一个web3provider的工具类，用来兼容MetaMask和TronLink，脱离UI框架（不是wagmi那种hooks），实现支持多链的钱包和合约交互，比如获取（初始化）provider，connect，getAccount，changeChain，transferToken，getContract，signMessage，signTransaction等等，相当于把ethers（和@metamask/sdk）和tronweb封装了一下。不过只满足了自己的项目需求，不是高大全的。

**Reply by jeasonstudio** [reply]
项目可以分享下吗，我们也欢迎能有多链支持相关的需求或 PR。

**Reply by alienzhangyw** [reply]
商业项目，目前不能公开，抱歉🫠
